server/src/server.ts

```diff
@@ -34,10 +40,9 @@ connection.onInitialize((params: InitializeParams) => {
     hasWorkspaceFolderCapability = !!(
         capabilities.workspace && !!capabilities.workspace.workspaceFolders
     );
-    hasDiagnosticRelatedInformationCapability = !!(
-        capabilities.textDocument &&
-        capabilities.textDocument.publishDiagnostics &&
-        capabilities.textDocument.publishDiagnostics.relatedInformation
+
+    hasDocumentLinkCapabilities = !!(
+        capabilities.textDocument && capabilities.textDocument.documentLink
     );
```

Comment:
In TypeScript ≥ 3.8 you could do the following:
```suggestion
    hasDocumentLinkCapabilities = !!capabilities.textDocument?.documentLink;
```

Reply:
voer ik door, voor die anderen lukt dat niet toch?
```
hasWorkspaceFolderCapability = !!(
        capabilities.workspace && !!capabilities.workspace.workspaceFolders
    );
````

Reply:
Volgens mij moet dat met het volgende kunnen:

```typescript
hasWorkspaceFolderCapability = !!capabilities.workspace?.workspaceFolders;
```

Reply:
done